components/persistence/varint/src/test/java/org/apache/polaris/persistence/varint/TestVarInt.java

```diff
@@ -0,0 +1,104 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(SoftAssertionsExtension.class)
+public class TestVarInt {
+  @InjectSoftAssertions SoftAssertions soft;
+
+  @Test
+  public void negative() {
+    var buf = ByteBuffer.allocate(9);
+    soft.assertThatIllegalArgumentException().isThrownBy(() -> VarInt.putVarInt(buf, -1L));
+    soft.assertThatIllegalArgumentException()
+        .isThrownBy(() -> VarInt.putVarInt(buf, Long.MIN_VALUE));
+  }
+
+  @ParameterizedTest
+  @MethodSource
+  public void varInt(long value, byte[] binary) {
+    var buf = ByteBuffer.allocate(9);
+    VarInt.putVarInt(buf, value);
+    soft.assertThat(buf.position()).isEqualTo(binary.length);
+    soft.assertThat(Arrays.copyOf(buf.array(), buf.position())).containsExactly(binary);
+    soft.assertThat(VarInt.varIntLen(value)).isEqualTo(binary.length);
+
+    var read = buf.duplicate().flip();
+    VarInt.skipVarInt(read);
+    soft.assertThat(read.position()).isEqualTo(binary.length);
+
+    if (value > Integer.MAX_VALUE) {
+      soft.assertThatIllegalArgumentException()
+          .isThrownBy(() -> VarInt.readVarInt(buf.duplicate().flip()))
+          .withMessageStartingWith("Out of range: ");
+      soft.assertThat(VarInt.readVarLong(buf.duplicate().flip())).isEqualTo(value);
+    } else {
+      soft.assertThat(VarInt.readVarInt(buf.duplicate().flip())).isEqualTo(value);
+      soft.assertThat(VarInt.readVarLong(buf.duplicate().flip())).isEqualTo(value);
+    }
+  }
+
+  @Test
+  public void notVarInt() {
+    var buf = new byte[] {-1, -1, -1, -1, -1, -1, -1, -1, -1, -1};
+    soft.assertThatIllegalArgumentException()
+        .isThrownBy(() -> VarInt.readVarInt(ByteBuffer.wrap(buf)));
+    soft.assertThatIllegalArgumentException()
+        .isThrownBy(() -> VarInt.skipVarInt(ByteBuffer.wrap(buf)));
+  }
+
+  static Stream<Arguments> varInt() {
+    return Stream.of(
+        // one byte
+        arguments(0L, new byte[] {0}),
+        arguments(1L, new byte[] {1}),
+        arguments(42L, new byte[] {42}),
+        arguments(127L, new byte[] {127}),
+        // 2 bytes
+        arguments(128L, new byte[] {(byte) 0x80, 1}),
+        // 21 bite -> 3 x 7 bits
+        arguments(0x1fffff, new byte[] {-1, -1, 127}),
+        // 28 bits -> 4 x 7 bits
+        arguments(0xfffffff, new byte[] {-1, -1, -1, 127}),
+        // 35 bits -> 5 x 7 bits
+        arguments(0x7ffffffffL, new byte[] {-1, -1, -1, -1, 127}),
+        // 42 bits -> 6 x 7 bits
+        arguments(0x3ffffffffffL, new byte[] {-1, -1, -1, -1, -1, 127}),
+        // 49 bits -> 7 x 7 bits
+        arguments(0x1ffffffffffffL, new byte[] {-1, -1, -1, -1, -1, -1, 127}),
+        // 56 bits -> 8 x 7 bits
+        arguments(0xffffffffffffffL, new byte[] {-1, -1, -1, -1, -1, -1, -1, 127}),
```

Comment:
might be worth having a test with zeros (and small numbers) in the middle :thinking:

Reply:
0 doesn't make sense - that terminates the varint.

Reply:
I meant zeros in the hex representation :)